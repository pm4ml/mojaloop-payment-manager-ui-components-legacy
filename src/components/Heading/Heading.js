import './Heading.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

function headerTag(size) {
  if (size >= 1 && size <= 6) {
    return `h${size}`;
  }
  return 'h3';
}

class Heading extends PureComponent {
  render() {
    const { size, children, style } = this.props;
    const Header = headerTag(parseInt(size, 10));
    return (
      <Header style={style} className="el">
        <span>{children}</span>
      </Header>
    );
  }
}

Heading.propTypes = {
  size: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  style: PropTypes.shape(),
  children: PropTypes.node,
};
Heading.defaultProps = {
  size: 3,
  style: undefined,
  children: undefined,
};

export default Heading;
