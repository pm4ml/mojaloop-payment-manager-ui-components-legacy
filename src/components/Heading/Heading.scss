h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0;
  margin-bottom: 2rem;
  span {
    font-family: Ubuntu, sans-serif;
    display: flex;
  }
}
h1 {
  line-height: 39px;
}
h2 {
  line-height: 36px;
}
h3 {
  line-height: 33px;
}
h4 {
  line-height: 30px;
}
h5 {
  line-height: 27px;
}
h6 {
  line-height: 24px;
}
