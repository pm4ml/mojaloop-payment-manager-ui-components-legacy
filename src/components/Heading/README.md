# Heading 

Renders a header - HTML tag `<h>`

### Proptypes
|   Name   |        Type        |        Description         | Optional | Values |   Default   |
|----------|--------------------|----------------------------|----------|--------|-------------|
| size     | `number`, `string` | describes the heading size |          |        | `3`         |
| style    | `object`           | describes the HTML style   |          |        | `undefined` |
| children | `node`             | describes the content      |          |        | `undefined` |