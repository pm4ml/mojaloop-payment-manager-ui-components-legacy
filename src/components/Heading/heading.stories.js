import React from 'react';

import Column from '../Column';
import Heading from './Heading';

export default {
  title: 'Heading',
  component: Heading,
};

export const Sizes = () => (
  <div>
    <Column>
      <Heading size="1"> Size 1 </Heading>
      <Heading size="2"> Size 2 </Heading>
      <Heading size="3"> Size 3 </Heading>
      <Heading size="4"> Size 4 </Heading>
      <Heading size="5"> Size 5 </Heading>
      <Heading size="6"> Size 6 </Heading>
    </Column>
  </div>
);
