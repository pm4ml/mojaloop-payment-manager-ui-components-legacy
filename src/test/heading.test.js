import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import Heading from '../components/Heading';

it('renders correct sizes', () => {
  [...Array(10).keys()].forEach(size => {
    const wrapper = shallow(<Heading size={size}> Text </Heading>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
