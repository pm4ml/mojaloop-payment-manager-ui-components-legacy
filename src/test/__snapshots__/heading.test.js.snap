// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct sizes 1`] = `
<h3
  className="el"
>
  <span>
     Text 
  </span>
</h3>
`;

exports[`renders correct sizes 2`] = `
<h1
  className="el"
>
  <span>
     Text 
  </span>
</h1>
`;

exports[`renders correct sizes 3`] = `
<h2
  className="el"
>
  <span>
     Text 
  </span>
</h2>
`;

exports[`renders correct sizes 4`] = `
<h3
  className="el"
>
  <span>
     Text 
  </span>
</h3>
`;

exports[`renders correct sizes 5`] = `
<h4
  className="el"
>
  <span>
     Text 
  </span>
</h4>
`;

exports[`renders correct sizes 6`] = `
<h5
  className="el"
>
  <span>
     Text 
  </span>
</h5>
`;

exports[`renders correct sizes 7`] = `
<h6
  className="el"
>
  <span>
     Text 
  </span>
</h6>
`;

exports[`renders correct sizes 8`] = `
<h3
  className="el"
>
  <span>
     Text 
  </span>
</h3>
`;

exports[`renders correct sizes 9`] = `
<h3
  className="el"
>
  <span>
     Text 
  </span>
</h3>
`;

exports[`renders correct sizes 10`] = `
<h3
  className="el"
>
  <span>
     Text 
  </span>
</h3>
`;
